app/src/test/java/nerd/tuxmobil/fahrplan/congress/search/SearchViewModelTest.kt

```diff
@@ -147,53 +207,78 @@ class SearchViewModelTest {
     inner class Query {
 
         @Test
-        fun `searchQuery matches passed string when OnSearchHistoryItemClick event is received`() =
+        fun `query matches passed string when OnSearchHistoryItemClick event is received`() =
             runTest {
-                val repository = createRepository()
-                val viewModel = createViewModel(repository)
+                val viewModel = createViewModel()
+
                 viewModel.onViewEvent(OnSearchHistoryItemClick("query"))
-                assertThat(viewModel.searchQuery).isEqualTo("query")
+
+                viewModel.uiState.test {
+                    assertThat(awaitItem().query).isEqualTo("query")
+                }
             }
 
         @Test
-        fun `searchQuery matches passed string when OnSearchQueryChange event is received`() {
-            val repository = createRepository()
-            val viewModel = createViewModel(repository)
+        fun `query matches passed string when OnSearchQueryChange event is received`() = runTest {
+            val viewModel = createViewModel()
+
             viewModel.onViewEvent(OnSearchQueryChange("query"))
-            assertThat(viewModel.searchQuery).isEqualTo("query")
+
+            viewModel.uiState.test {
+                assertThat(awaitItem().query).isEqualTo("query")
+            }
         }
 
         @Test
-        fun `searchQuery is empty string when OnSearchQueryClear event is received`() {
-            val repository = createRepository()
-            val viewModel = createViewModel(repository)
+        fun `query is empty string when OnSearchQueryClear event is received`() = runTest {
+            val viewModel = createViewModel()
+
             viewModel.onViewEvent(OnSearchQueryClear)
-            assertThat(viewModel.searchQuery).isEmpty()
+
+            viewModel.uiState.test {
+                assertThat(awaitItem().query).isEmpty()
+            }
         }
 
     }
 
-    private fun createRepository(
+    private fun createSearchHistoryManager(): SearchHistoryManager {
+        return SearchHistoryManager(InMemorySearchRepository())
+    }
+
+    private fun createViewModel(
+        searchHistoryManager: SearchHistoryManager = createSearchHistoryManager(),
         sessionsFlow: Flow<List<Session>> = flowOf(emptyList()),
```

Comment:
Please rename to `sessions` so we don't have to handle the "type" in the future when it might change. Same below and above in the comment.